test/x509aux.c

```diff
@@ -173,59 +159,15 @@ static int test_certs(BIO *fp)
     }
 
     /* Some other PEM read error */
-    print_errors();
     return 0;
 }
 
-int main(int argc, char *argv[])
+int test_main(int argc, char *argv[])
 {
-    BIO *bio_err;
-    const char *p;
-    int ret = 1;
-
-    progname = argv[0];
-    if (argc < 2) {
-        test_usage();
-        EXIT(ret);
-    }
-
-    bio_err = BIO_new_fp(stderr, BIO_NOCLOSE | BIO_FP_TEXT);
-
-    p = getenv("OPENSSL_DEBUG_MEMORY");
-    if (p != NULL && strcmp(p, "on") == 0)
-        CRYPTO_set_mem_debug(1);
-    CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_ON);
-
-    argc--;
-    argv++;
-
-    while (argc >= 1) {
-        BIO *f = BIO_new_file(*argv, "r");
-        int ok;
-
-        if (f == NULL) {
-            fprintf(stderr, "%s: Error opening cert file: '%s': %s\n",
-                    progname, *argv, strerror(errno));
-            EXIT(ret);
-        }
-        ret = !(ok = test_certs(f));
-        BIO_free(f);
-
-        if (!ok) {
-            printf("%s ERROR\n", *argv);
-            ret = 1;
-            break;
-        }
-        printf("%s OK\n", *argv);
-
-        argc--;
-        argv++;
-    }
+    if (argc < 2)
+        TEST_error("usage: %s certfile...", argv[0]);
 
-#ifndef OPENSSL_NO_CRYPTO_MDEBUG
-    if (CRYPTO_mem_leaks(bio_err) <= 0)
-        ret = 1;
-#endif
-    BIO_free(bio_err);
-    EXIT(ret);
+    files = &argv[1];
+    ADD_ALL_TESTS(test_certs, argc - 1);
```

Comment:
it says "zero test cases" and then reports passed.  I changed it to report failure.  Not a big deal.